MIP/mip-25/README.md

```diff
@@ -0,0 +1,95 @@
+5. Updates to the framework would use a Move script signed by the approved proxy account to acquire signing priviledges of aptos framework during the scripts execution
+6. Ideally the controlling account would be multisig and this would be a strict requirement for the proxy module.  For test purposes this would be relaxed to a single account
+
+## Reference Implementation
+```rust
+module aptos_framework::proxy {
+    use std::signer;
+    use aptos_framework::account::{create_signer_with_capability, SignerCapability};
+    use aptos_framework::event;
+    use aptos_framework::system_addresses;
+    use aptos_framework::system_addresses::is_aptos_framework_address;
+    friend aptos_framework::genesis;
+
+    struct ProxyController has key {
+        controller: address,
+        aptos_framework_signer_cap: SignerCapability,
+    }
+
+    const EINVALID_CONTROLLER : u64 = 0x1;
+
+    #[event]
+    /// An event triggered when the controller is updated
+    struct ControllerUpdated has store, drop {
+        old_controller: address,
+        new_controller: address,
+    }
+
+    /// Only called during genesis.
+    public(friend) fun initialize(aptos_framework: &signer, aptos_framework_signer_cap: SignerCapability) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        move_to(aptos_framework, ProxyController {
+            controller: @aptos_framework,
+            aptos_framework_signer_cap,
+        })
+    }
+
+    /// Update the controller of the proxy
+    /// Aborts if signed by anyone else except the controller or @aptos_framework
+    /// Emits an event on updating the controller
+    public fun update_controller(signer: &signer, new_controller: address) acquires ProxyController {
+        let proxy_controller = borrow_global_mut<ProxyController>(@aptos_framework);
+
+        assert!(
+            proxy_controller.controller == signer::address_of(signer) ||
+            is_aptos_framework_address(signer::address_of(signer)),
+            EINVALID_CONTROLLER);
+
+        let old_controller = proxy_controller.controller;
+        proxy_controller.controller = new_controller;
+
+        event::emit(
+            ControllerUpdated {
+                old_controller,
+                new_controller,
+            },
+        );
+    }
+
+    #[view]
+    /// Return the controller of the proxy
+    public fun get_controller(): address acquires ProxyController {
+        borrow_global<ProxyController>(@aptos_framework).controller
+    }
+
+    /// Delegate aptos framework's signing cap to proxy
+    public fun delegate_to_proxy(controller: &signer) : signer acquires ProxyController {
+        let proxy_controller = borrow_global<ProxyController>(@aptos_framework);
+        assert!(signer::address_of(controller) == proxy_controller.controller, EINVALID_CONTROLLER);
+        create_signer_with_capability(&proxy_controller.aptos_framework_signer_cap)
+    }
+}
+```
```

Comment:
If you can provide an example usage of this module and perhaps a sequence diagram for its usage with the bridge, that would be great.